completions/make

```diff
@@ -157,19 +157,65 @@ _make()
             fi
         done
 
-        # recognise that possible completions are only going to be displayed
-        # so only the base name is shown
+        # recognise that possible completions are only going to be displayed so
+        # only the base name is shown.
+        #
+        # Note: This is currently turned off because the test suite of
+        # bash-completion conflicts with it; it uses "set show-all-if-ambiguous
+        # on" (causing COMP_TYPE == 37) to retrieve the action completion
+        # results, and also the compact form with only the basenames is not
+        # essentially needed.  To re-enable it, please uncomment the following
+        # if-statement.
         local mode=--
-        if ((COMP_TYPE != 9)); then
-            mode=-d # display-only mode
-        fi
+        # if ((COMP_TYPE != 9 && COMP_TYPE != 37 && COMP_TYPE != 42)); then
+        #     mode=-d # display-only mode
+        # fi
 
         local IFS=$' \t\n' script=$(_make_target_extract_script $mode "$cur")
         COMPREPLY=($(LC_ALL=C \
             $1 -npq __BASH_MAKE_COMPLETION__=1 \
             ${makef+"${makef[@]}"} "${makef_dir[@]}" .DEFAULT 2>/dev/null |
             command sed -ne "$script"))
 
+        local prefix=$cur
+        [[ $mode == -d ]] && prefix=
+        if ((${#COMPREPLY[@]} > 0)); then
+            local -A processed paths
+            local target
+            for target in "${COMPREPLY[@]}"; do
+                local path=${target%/}
+                [[ ${processed[$path]+set} ]] && continue
+                processed[$path]=$target
+
+                while
+                    if [[ ! ${paths[$path]+set} ]]; then
+                        ((paths[\$path] = 1))
+                    else
+                        ((paths[\$path]++))
+                    fi
+                    [[ $path == "$prefix"*/* ]]
+                do
+                    path=${path%/*}
+                done
+            done
+
+            COMPREPLY=()
+            local nreply=0 target
```

Comment:
`target` was already defined above. 

```suggestion
            local nreply=0
```

Reply:
Thank you! I have removed it as a part of commit 0ef11ca15b98abb47394f7c39208812239a6c379.